test/Broadway/EventSourcing/EventSourcingRepositoryTest.php

```diff
@@ -78,6 +81,34 @@ public function it_throws_an_exception_if_the_static_method_does_not_exist()
         $repository->load('y0l0');
     }
 
+    /**
+     * @test
+     */
+    public function it_publishes_decorated_events()
+    {
+        $projector = new TestMetadataPublishedProjector();
+        $this->eventBus->subscribe($projector);
+
+        $repository = new EventSourcingRepository(
+            $this->eventStore,
+            $this->eventBus,
+            '\Broadway\EventSourcing\TestEventSourcedAggregate',
+            new PublicConstructorAggregateFactory(),
+            array(new MetadataEnrichingEventStreamDecorator(array(new TestDecorationMetadataEnricher())))
+        );
+
+        $aggregate = $this->createAggregate();
+        $aggregate->apply(new DidNumberEvent(42));
+        $repository->add($aggregate);
+
+        /** @var Metadata $metadata */
```

Comment:
We don't add these comments in the middle of code. Feel free to add a doc block to the projector (that should fix it as well?).